exporter/azuredataexplorerexporter/adx_exporter.go

```diff
@@ -0,0 +1,281 @@
+	// Expect that this mapping is already existent
+	if refoption := getMappingRef(config, telemetrydatatype); refoption != nil {
+		ingestoptions = append(ingestoptions, refoption)
+	}
+	// The exporter could be configured to run in either modes. Using managedstreaming or batched queueing
+	if strings.ToLower(config.IngestionType) == managedingesttype {
+		mi, err := createManagedStreamingIngestor(config, metricclient, tablename)
+		if err != nil {
+			return nil, err
+		}
+		ingestor = mi
+		err = nil
+	} else {
+		qi, err := createQueuedIngestor(config, metricclient, tablename)
+		if err != nil {
+			return nil, err
+		}
+		ingestor = qi
+		err = nil
+	}
+	return &adxDataProducer{
+		client:        metricclient,
+		ingestoptions: ingestoptions,
+		ingestor:      ingestor,
+		logger:        logger,
+	}, nil
+}
+
+/**
+Common functions that are used by all the 3 parts of OTEL , namely Traces , Logs and Metrics
+*/
+
+/* Fetchs the coresponding ingetionRef if the mapping is provided*/
+func getMappingRef(config *Config, telemetrydatatype int) ingest.FileOption {
+	switch telemetrydatatype {
+	case metricstype:
+		if !isEmpty(config.OTELMetricTableMapping) {
+			return ingest.IngestionMappingRef(config.OTELMetricTableMapping, ingest.JSON)
+		}
+	case tracestype:
+		if !isEmpty(config.OTELTraceTableMapping) {
+			return ingest.IngestionMappingRef(config.OTELTraceTableMapping, ingest.JSON)
+		}
+	case logstype:
+		if !isEmpty(config.OTELLogTableMapping) {
+			return ingest.IngestionMappingRef(config.OTELLogTableMapping, ingest.JSON)
+		}
+	}
+	return nil
+}
+
+func buildAdxClient(config *Config) (*kusto.Client, error) {
+	authorizer := kusto.Authorization{
+		Config: auth.NewClientCredentialsConfig(config.ClientId,
+			config.ClientSecret, config.TenantId),
+	}
+	client, err := kusto.New(config.ClusterName, authorizer)
+	return client, err
+}
+
+// Depending on the table , create seperate ingestors
+func createManagedStreamingIngestor(config *Config, adxclient *kusto.Client, tablename string) (*ingest.Managed, error) {
+	ingestor, err := ingest.NewManaged(adxclient, config.Database, tablename)
+	return ingestor, err
+}
+
+// A queued ingestor in case that is provided as the config option
+func createQueuedIngestor(config *Config, adxclient *kusto.Client, tablename string) (*ingest.Ingestion, error) {
+	ingestor, err := ingest.New(adxclient, config.Database, tablename)
+	return ingestor, err
+}
+
+func getScopeMap(sc pcommon.InstrumentationScope) map[string]interface{} {
+	scopeMap := make(map[string]interface{}, 2)
+
+	if sc.Name() != "" {
+		scopeMap[scopename] = sc.Name()
+	}
+	if sc.Version() != "" {
+		scopeMap[scopeversion] = sc.Version()
+	}
+
+	return scopeMap
+}
+
+func getTableName(config *Config, telemetrydatatype int) string {
+	switch telemetrydatatype {
+	case metricstype:
+		return config.OTELMetricTable
+	case logstype:
+		return config.OTELLogTable
+	case tracestype:
+		return config.OTELTraceTable
+	}
+	return "unknown"
```

Comment:
It's better to return and check err, or you'll end up ingestion to a table named "unknown"